plotly/tools.py

```diff
@@ -1670,6 +1722,321 @@ def create_streamline(x, y, u, v,
                              mode='lines', **kwargs)
         return streamline
 
+    @staticmethod
+    def create_ohlc(open, high, low, close, direction, **kwargs):
+
+        """
+        Returns data for an ohlc chart
+
+        :param (list) open: opening values
+        :param (list) high: high values
+        :param (list) low: low values
+        :param (list) close: closing values
+        :param (string) direction: direction can be 'increasing' or
+            'decreasing'. When the direction is 'increasing', the returned data
+            consists of all units where the close value is greater than the
+            corresponding open value, and when the direction is 'decreasing',
+            the returned data consists of all units where the close value is
+            less than or equal to the corresponding open value.
+        :param (class) kwargs: kwargs passed through plotly.graph_objs.Scatter
+            for more information on valid kwargs call
+            help(plotly.graph_objs.Scatter)
+
+        :rtype (trace): returns data for ohlc increasing units or decreasing
+            units.
+
+        Example 1: Plot ohlc chart
+        ```
+        import plotly.plotly as py
+        import plotly.tools as tls
+        from plotly.graph_objs import *
+
+        # Add data
+        open_data = [33.0, 33.3, 33.5, 33.0, 34.1]
+        high_data = [33.1, 33.3, 33.6, 33.2, 34.8]
+        low_data = [32.7, 32.7, 32.8, 32.6, 32.8]
+        close_data = [33.0, 32.9, 33.3, 33.1, 33.1]
+
+        # Create ohlc increasing units
+        ohlc_increase = tls.TraceFactory.create_ohlc(open_data, high_data,
+                                                     low_data, close_data,
+                                                     direction='increasing')
+
+        # Create ohlc decreasing units
+        ohlc_decrease = tls.TraceFactory.create_ohlc(open_data, high_data,
+                                                     low_data, close_data,
+                                                     direction='decreasing')
```

Comment:
@chriddyp if we combine both increasing and decreasing into one call is there an easier way to pass the kwargs for each trace than doing something like: 
```
kwargs.setdefault('name', ('Increasing', 'Decreasing'))
kwargs.setdefault('line', {'color': ('rgb(44, 160, 44)', 'rgb(214, 39, 40)')})
kwargs.setdefault('text', (text_increase, text_decrease))

ohlc_icrease = Scatter(x=flat_increase_x,
                                     y=flat_increase_y,
                                     mode='lines',
                                     name=kwargs['name'][0],
                                     text=kwargs['text'][0],
                                     line=Line(color=kwargs['line']['color'][0])
                        )

ohlc_decrease = Scatter(x=flat_decrease_x,
                                        y=flat_decrease_y,
                                        mode='lines',
                                        name=kwargs['name'][1],
                                        text=kwargs['text'][1],
                                        line=Line(color=kwargs['line']['color'][1])
                         )
```